core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -105,11 +110,7 @@ public void run() {
           }, Configuration.getInt(PropertyKey.USER_HEARTBEAT_INTERVAL_MS),
           Configuration.getInt(PropertyKey.USER_HEARTBEAT_INTERVAL_MS), TimeUnit.MILLISECONDS);
 
-      try {
-        sessionHeartbeat();
-      } catch (InterruptedException e) {
-        throw Throwables.propagate(e);
-      }
+      NUM_ACTIVE_SESSIONS.incrementAndGet();
```

Comment:
Would it be possible to have the ctor clean up the heartbeat on an exception? Then we can avoid a lot of the logic below.